collections/users.go

```diff
@@ -0,0 +1,67 @@
+package collections
+
+import (
+	"context"
+
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/textile/v2/badgerdb"
+	"github.com/textileio/textile/v2/model"
+	"github.com/textileio/textile/v2/mongodb"
+)
+
+type Users struct {
+	hub bool
+	m   mongodb.Users
+	b   badgerdb.Users
+}
+
+type UsersOptions func(*Users)
+
+func WithMongoUsersOpts(m mongodb.Users) UsersOptions {
+	return func(i *Users) {
+		i.m = m
+	}
+}
+
+func WithBadgerUsersOpts(b badgerdb.Users) UsersOptions {
+	return func(i *Users) {
+		i.b = b
+	}
+}
+
+func NewUsers(_ context.Context, hub bool, opts ...UsersOptions) (*Users, error) {
+	u := &Users{
+		hub: hub,
+	}
+
+	for _, opt := range opts {
+		opt(u)
+	}
+
+	return u, nil
+}
+
+func (u *Users) UpdatePowInfo(ctx context.Context, key thread.PubKey, powInfo *model.PowInfo) (*model.User, error) {
```

Comment:
@jsonsivar 

What do you think about creating even a small interface like:

```
type NotAGoodName interface {
    UpdatePowInfo(ctx, key, powInfo)
...
}
```

And using it in the `type IPNSKeys struct`?

(with proper signatures, of course)